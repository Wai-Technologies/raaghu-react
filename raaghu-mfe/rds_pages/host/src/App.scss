//@import "raaghu-react-themes/build/styles/default.css";
// Please dont remove below comment line Adding raaghu-theme in local side.
@import "../../../../raaghu-react-themes/src/styles/default.scss";

.main-body {
    margin-top: 29px;
    position: relative;
}
