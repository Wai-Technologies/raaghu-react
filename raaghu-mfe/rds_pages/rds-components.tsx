export * from "../../raaghu-components/src/index";
