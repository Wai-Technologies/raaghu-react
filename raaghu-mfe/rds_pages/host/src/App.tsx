import React, { Suspense } from "react";
import "bootstrap/dist/js/bootstrap.bundle.min.js";
import "./App.scss";
import AuthHigherComponent from "./AuthHigherComponent";

const App = () => {
    return (
        <Suspense>
            <AuthHigherComponent></AuthHigherComponent>
        </Suspense>
    );
};

export default App;
