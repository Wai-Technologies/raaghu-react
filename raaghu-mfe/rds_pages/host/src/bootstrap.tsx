import App from "./App";
import ReactDOM from "react-dom/client";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../../libs/public.api";
import { Provider } from "react-redux";

(ReactDOM as any).createRoot(document.getElementById("root")).render(
    <Provider store={store}>
        <div>
            <BrowserRouter>
                <App />
            </BrowserRouter>
        </div>
    </Provider>
);

