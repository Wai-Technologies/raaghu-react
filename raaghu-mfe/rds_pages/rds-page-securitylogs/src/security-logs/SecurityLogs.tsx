import React, { Suspense, useEffect, useState } from "react";
import {RdsButton, RdsInput, RdsDatePicker} from "../../../../../raaghu-elements/src";
import {
    useAppDispatch,
    useAppSelector,
} from "../../../../libs/state-management/hooks";
import { fetchSecurityLogs } from "../../../../libs/state-management/security-logs/security-logs-slice";
import { RdsCompAlertPopup, RdsCompDatatable } from "../../../rds-components";
import { useTranslation } from "react-i18next";


const SecurityLogs = () => {
    const { t } = useTranslation();
    const data = useAppSelector((state) => state.persistedReducer.securityLogs);
    const [securityLogsData, setData] = useState<any>([]);
    const [skipCount, setSkipCount] = useState<number>(0);
    const [recordsPerPage, setRecordsPerPage] = useState<number>(10);
    const [totalRecords, setTotalRecords] = useState<number | null>(0);
    const [currentDate, setCurrentDate] = useState(new Date());
    const [tomorrowDate, setTomorrowDate] = useState(new Date());
    const [recordsFound, setRecordsFound] = useState(true);
    const [maxResultCount, setMaxResultCount] = useState<number>(10);

    const tableHeadersForExcel = [
        "Time",
        "Actions",
        "IP Address",
        "Browser/Os",
        "Application",
        "Identity",
        "Users",
    ];

    const tableHeaders = [
        {
            displayName: t("Time"),
            key: "creationTime",
            datatype: "date",
            sortable: true,
        },
        {
            displayName: t("Actions"),
            key: "action",
            datatype: "text",
            sortable: true,
        },
        {
            displayName: t("IP Address"),
            key: "clientIpAddress",
            datatype: "text",
            sortable: true,
        },
        {
            displayName: t("Browser/Os"),
            key: "browserInfo",
            datatype: "tooltip",
            sortable: true,
            dataLength: 35,
        },
        {
            displayName: t("AbpAccount.MySecurityLogs:Application"),
            key: "applicationName",
            datatype: "text",
            sortable: true,
        },
        {
            displayName: t("AbpAccount.MySecurityLogs:Identity"),
            key: "identity",
            datatype: "text",
            sortable: true,
        },
        {
            displayName: t("AbpIdentity.Users"),
            key: "userName",
            datatype: "text",
            sortable: true,
        },
    ];

    useEffect(() => {
        setCurrentDate(new Date());
    }, []);

    const formatDateString = (date: Date) => {
        return `${date.getFullYear()}-${(date.getMonth() + 1).toString().padStart(2, '0')}-${date.getDate().toString().padStart(2, '0')}`;
    }

    const formattedCurrentDate = formatDateString(currentDate);
    tomorrowDate.setDate(currentDate.getDate() + 1);
    const formattedTomorrowDate = formatDateString(tomorrowDate);

    let [selectFilterValue, setSelectFilterValue] = useState({
        action: "",
        userName: "",
        startTime: formattedCurrentDate,
        endTime: formattedTomorrowDate
    });




    function downloadcsv() {
        const dataToDownload = securityLogsData.map((item: any) => [
            item?.creationTime,
            item?.action,
            item?.clientIpAddress,
            item?.browserInfo.replace(/,/g, '-'),
            item?.applicationName,
            item?.identity,
            item?.userName,
        ]);

        const csvStrings = [
            tableHeadersForExcel.map(header => t(header)).join(',') + '\r\n',
            ...dataToDownload.map((row: any) => row.join(',') + '\r\n'),
        ].join('');

        const blob = new Blob([csvStrings], { type: 'text/csv' });
        const url = window.URL.createObjectURL(blob);
        const a = document.createElement('a');
        a.href = url;
        a.download = 'securityLogs.csv';
        document.body.appendChild(a);
        a.click();
        window.URL.revokeObjectURL(url);
    }

    const dispatch = useAppDispatch();
    const securityLogsParamsData = {
        action: selectFilterValue.action,
        userName: selectFilterValue.userName,
        startTime: selectFilterValue.startTime,
        endTime: selectFilterValue.endTime
    };

    const securityLogsTableData = {
        action: "",
        identity: "",
        browserInfo: "",
        userName: "",
        applicationName: "",
        clientIpAddress: "",
        startDate: "",
        endDate: ""
    };


    useEffect(() => {
        dispatch(fetchSecurityLogs({ skipCount: skipCount, maxResultCount: maxResultCount, startTime: formattedCurrentDate, endTime: formattedTomorrowDate }) as any);
    }, [dispatch, skipCount, maxResultCount])

    useEffect(() => {
        if (data?.securityLogs) {
            const tableDataShow = data.securityLogs?.items.map((item: any) => {
                return {
                    id: item?.id,
                    startTime: item?.startTime,
                    endTime: item?.endTime,
                    creationTime: item.creationTime,
                    action: item?.action,
                    clientIpAddress: item?.clientIpAddress,
                    browserInfo: item?.browserInfo,
                    applicationName: item?.applicationName,
                    identity: item?.identity,
                    userName: item?.userName,
                };
            });
            setData(tableDataShow);
            setTotalRecords(data?.securityLogs?.totalCount);
        }
    }, [data?.securityLogs]);



    const onActionFilter = (event: any) => {
        setSelectFilterValue({
            ...selectFilterValue,
            action: event.target.value,
        });
        if (event.target.value.length > 5) {
            dispatch(fetchSecurityLogs({ ...securityLogsParamsData, action: event.target.value }) as any);
        }
        if (securityLogsData.length > 0) {
            setRecordsFound(false);
        }
        else {
            setRecordsFound(true);
        }
    };

    const onUserNameFilter = (event: any) => {
        setSelectFilterValue({
            ...selectFilterValue,
            userName: event?.target?.value,
        });
        if (event.target.value.length >= 3) {
            dispatch(fetchSecurityLogs({ ...securityLogsParamsData, userName: event.target.value }) as any);
        }
        else {
            dispatch(fetchSecurityLogs({ skipCount: skipCount, maxResultCount: recordsPerPage }) as any);
        }
        if (securityLogsData.length > 0) {
            setRecordsFound(false);
        }
        else {
            setRecordsFound(true);
        }
        // setRecordsFound(securityLogsData.length > 0);
    };

    const formatDate = (inputDate: any) => {
        const dateObj = new Date(inputDate);
        const year = dateObj.getFullYear();
        const month = String(dateObj.getMonth() + 1).padStart(2, '0');
        const day = String(dateObj.getDate()).padStart(2, '0');
        return `${year}-${month}-${day}`;
    }

    const onDateSelectFilter = (startDate: any, endDate: any) => {
        const [start, end] = startDate;

        const currentDate = new Date();
        const formattedCurrentDate = formatDate(currentDate);

        let formattedStart = formatDate(start);
        let formattedEnd = formatDate(end);

        if (formattedStart === formattedCurrentDate) {
            const nextDay = new Date(currentDate);
            nextDay.setDate(nextDay.getDate() + 1);
            formattedEnd = formatDate(nextDay);
        }

        const yesterday = new Date(currentDate);
        yesterday.setDate(yesterday.getDate() - 1);
        if (formattedStart === formatDate(yesterday)) {
            const dayBeforeYesterday = new Date(yesterday);
            dayBeforeYesterday.setDate(dayBeforeYesterday.getDate() - 1);
            formattedStart = formatDate(dayBeforeYesterday);
            formattedEnd = formatDate(yesterday);
        }

        setSelectFilterValue({
            ...selectFilterValue,
            startTime: formattedStart,
            endTime: formattedEnd
        });
        dispatch(fetchSecurityLogs({ ...securityLogsParamsData, startTime: formattedStart, endTime: formattedEnd }) as any);
    }



    const paginationHandler = (currentPage: number, recordsPP: number) => {
        const skipCount = recordsPP * (currentPage - 1);
        setSkipCount(skipCount);
        setMaxResultCount(recordsPP);
    };

    return (
        <Suspense>
            <div className="container-fluid p-0 m-0">
                <div className="row">
                    <div className="col-md-12">
                        <div className="card px-4 py-4 card-full-stretch rounded-0 border-0">
                            <div className="card-body p-0">

                                <div className="row align-items-center">
                                    <div className="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-3 mb-2 mt-2">
                                        <RdsDatePicker
                                            // DatePickerLabel={t(" Date ") || ""}
                                            type="advanced"
                                            // selectedDate={selectFilterValue.creationTime}
                                            customDate={onDateSelectFilter}
                                            isDropdownOpen={false}
                                        ></RdsDatePicker>
                                    </div>
                                    <div className="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-3 mb-2">
                                        <RdsInput
                                            inputType="text"
                                            // label={t("Actions ") || ""}
                                            placeholder="Search by Actions"
                                            labelPosition="top"
                                            value={selectFilterValue.action}
                                            size="small"
                                            onChange={onActionFilter}
                                        ></RdsInput>
                                    </div>
                                    <div className="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-3 mb-3">
                                        <RdsInput
                                            inputType="text"
                                            // label={t("Users ") || ""}
                                            placeholder="Search by Users"
                                            labelPosition="top"
                                            value={selectFilterValue.userName}
                                            size="small"
                                            onChange={onUserNameFilter}
                                        ></RdsInput>
                                    </div>
                                    <div className="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-0 mb-md-2 mb-3 d-xxl-flex d-xl-flex d-lg-flex justify-content-end">
                                        {recordsFound && (
                                            <>
                                                <RdsButton
                                                    type="button"
                                                    colorVariant="primary"
                                                    label={t("") || ""}
                                                    isOutline={true}
                                                    icon="downloads"
                                                    iconHeight="16px"
                                                    size="small"
                                                    iconFill={true}
                                                    iconStroke={true}
                                                    iconWidth="16px"
                                                    iconColorVariant="light"
                                                    onClick={downloadcsv}
                                                    tooltip={true}
                                                    tooltipTitle={"Download"}
                                                    tooltipPlacement="bottom"
                                                />
                                                <RdsCompAlertPopup
                                                    alertConfirmation="Download Security Log Files ?"
                                                    alertID="alert_popup"
                                                    cancelButtonLabel="Cancel"
                                                    colorVariant="success"
                                                    deleteButtonLabel="Yes"
                                                    messageAlert=" "
                                                    iconUrl="download"
                                                    data-bs-target="#alert_popup"
                                                    data-bs-toggle="modal"
                                                />
                                            </>

                                        )}
                                    </div>
                                </div>

                                <div className="row">
                                    <div className="col-md-12">
                                        <RdsCompDatatable
                                            tableHeaders={tableHeaders}
                                            tableData={securityLogsData}
                                            recordsPerPageSelectListOption={true}
                                            pagination={true}
                                            totalRecords={totalRecords}
                                            recordsPerPage={recordsPerPage}
                                            illustration={true}
                                            noDataheaderTitle={t("No Records Available") || ""}
                                            onPaginationHandler={paginationHandler}
                                        ></RdsCompDatatable>

                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
            <></>
        </Suspense>
    );
};

export default SecurityLogs;


