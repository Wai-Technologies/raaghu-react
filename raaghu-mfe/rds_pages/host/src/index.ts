import("./bootstrap");
import "../../../web.config";
