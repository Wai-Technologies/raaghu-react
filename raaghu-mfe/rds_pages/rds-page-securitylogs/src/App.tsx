import React, { Suspense } from "react";
import SecurityLogs from "./security-logs/SecurityLogs";

const App = () => (
    <Suspense>
        <SecurityLogs></SecurityLogs>
    </Suspense>
);

export default App;
