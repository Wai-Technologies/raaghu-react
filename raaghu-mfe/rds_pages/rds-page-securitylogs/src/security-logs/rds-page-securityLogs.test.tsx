import React from "react";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import { Provider, useDispatch } from "react-redux";
import { store } from "../../../../libs/state-management";
import SecurityLogs from "./SecurityLogs";
import { RdsButton, RdsDatePicker, RdsInput } from "../../../../../raaghu-elements/src";
import { RdsCompAlertPopup, RdsCompDatatable } from "../../../rds-components";

jest.mock("react-i18next", () => ({
    useTranslation: () => ({
      t: (key: string) => key,
    }),
    initReactI18next: { // add this line
      type: "3rdParty",
      init: () => {},
    },
  }));
jest.mock("react-router-dom", () => ({
    useNavigate: () => jest.fn(),
}));

global.location = {
    pathname: "../../../../libs/proxy/core",
  } as any;
  
  jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  }));
  jest.mock("../../../../../raaghu-components/src/index.ts", () => ({
    ...jest.requireActual("../../../../../raaghu-components/src/index.ts"),
  }));
  jest.mock("../../../../../raaghu-elements/src/index.ts", () => ({
    ...jest.requireActual("../../../../../raaghu-elements/src/index.ts"),
  }));
  jest.mock("../../../../libs/state-management/index.ts", () => ({
    store: {
      getState: jest.fn().mockReturnValue({
        persistedReducer: {
          configureStore: {
            configuration: {
              auth: {
                grantedPolicies: {},
              },
            },
          },
        },
      }),
      dispatch: jest.fn(),
      subscribe: jest.fn(),
    },
  }));
  
  // Mock the rds-elements and rds-components
  jest.mock("../../../../../raaghu-elements/src", () => ({
    RdsButton: jest.fn(),
    RdsOffcanvas: jest.fn(),
    RdsAlert: jest.fn(),
    RdsSearch: jest.fn(),
    RdsModal: jest.fn(),
    RdsLabel : jest.fn(),
    RdsInput : jest.fn(),
    RdsIcon : jest.fn(),
    RdsCheckbox : jest.fn(),
    RdsSelectList : jest.fn(),
    RdsBadge : jest.fn(),
    RdsFabMenu : jest.fn(),
    RdsNavtabs : jest.fn(),
    RdsDatePicker : jest.fn(),
  })); 
  
  jest.mock("../../../../../raaghu-components/src", () => ({
    RdsCompApiScopeBasicResource: jest.fn(),
    RdsCompDatatable: jest.fn(),
    RdsCompAlertPopup: jest.fn(),
    RdsOffcanvas:jest.fn(),
    RdsCompForgotPassword: jest.fn(),
    RdsCompFormsEmail: jest.fn(),
    RdsCompIconList: jest.fn(),
    RdsCompNewLanguage: jest.fn(),
    RdsCompWebsiteLog: jest.fn(),
    RdsCompCache: jest.fn(),
  }));
  
  jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useDispatch: jest.fn(),
  }));
  
  // Mock localStorage in your test setup
  const localStorageMock: Storage = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    clear: jest.fn(),
    length: 0, // Add length property
    key: jest.fn(), // Add key method
  };
  
  global.localStorage = localStorageMock;
  describe("Security-Logs Page", () => {
it("renders Security-Logs page correctly", () => {
    const mockDispatch = jest.fn();
    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
    render(
        <Provider store={store}>
            <SecurityLogs />
        </Provider>
    );
    document.querySelector('[placeholder="Search by Actions"]');
    document.querySelector('[placeholder="Search by Users"]');
});

it("Check security Logs page with data", () => {
    const onDateSelectFilter = jest.fn();
    const onActionFilter = jest.fn();
    const onUserNameFilter = jest.fn();
    const downloadcsv = jest.fn();
    const tableHeaders = { displayName: "Header", key: "key", datatype: "string", width: "width" };
    const securityLogsData = { key: "value" };
    const paginationHandler = jest.fn();
    const mockDispatch = jest.fn();
    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
   const security = render(
        <Provider store={store}>
            <SecurityLogs />
            <RdsDatePicker isDropdownOpen={false} customDate={onDateSelectFilter}/>
            <RdsInput placeholder="Search by Actions" onChange={onActionFilter}/>
            <RdsInput placeholder="Search by Users" onChange={onUserNameFilter}/>
            <RdsButton onClick={downloadcsv}/>
            <RdsCompAlertPopup alertID={"alert_popup"} alertConfirmation="Download Security Log Files ?"/>
            <RdsCompDatatable tableHeaders={[tableHeaders]} tableData={[securityLogsData]} pagination={false}  onPaginationHandler={paginationHandler}/>
        </Provider>
    );
    expect(security).toMatchSnapshot();
});

afterEach(() => {
    jest.clearAllMocks(); // Clear mock calls between tests
    // Reset localStorage to a new object
    global.localStorage = {
      getItem: jest.fn(),
      setItem: jest.fn(),
      removeItem: jest.fn(),
      clear: jest.fn(),
      length: 0,
      key: jest.fn(),
    };
  });
  })