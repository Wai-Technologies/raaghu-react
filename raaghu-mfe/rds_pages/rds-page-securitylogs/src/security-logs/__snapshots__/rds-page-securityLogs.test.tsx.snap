// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security-Logs Page Check security Logs page with data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid p-0 m-0"
      >
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <div
              class="card px-4 py-4 card-full-stretch rounded-0 border-0"
            >
              <div
                class="card-body p-0"
              >
                <div
                  class="row align-items-center"
                >
                  <div
                    class="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-3 mb-2 mt-2"
                  />
                  <div
                    class="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-3 mb-2"
                  />
                  <div
                    class="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-3 mb-3"
                  />
                  <div
                    class="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-0 mb-md-2 mb-3 d-xxl-flex d-xl-flex d-lg-flex justify-content-end"
                  />
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-md-12"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid p-0 m-0"
    >
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="card px-4 py-4 card-full-stretch rounded-0 border-0"
          >
            <div
              class="card-body p-0"
            >
              <div
                class="row align-items-center"
              >
                <div
                  class="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-3 mb-2 mt-2"
                />
                <div
                  class="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-3 mb-2"
                />
                <div
                  class="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-3 mb-3"
                />
                <div
                  class="col-xxl-3 col-xl-3 col-lg-3 col-md-6 col-12 mb-lg-0 mb-md-2 mb-3 d-xxl-flex d-xl-flex d-lg-flex justify-content-end"
                />
              </div>
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
